src/wms/src/main/java/org/geoserver/wms/capabilities/DimensionHelper.java

```diff
@@ -493,9 +493,13 @@ protected String getNumberRepresentation(
             elevationMetadata = buff.toString();
         } else if (DimensionPresentation.DISCRETE_INTERVAL == dimension.getPresentation()) {
             final NumberRange range = getMinMaxZInterval(values);
-            buff.append(range.getMinimum());
+            final Class<?> typeBinding = values.first().getClass();
+            final boolean isDecimal = isDecimal(typeBinding);
+            if (isDecimal) buff.append(range.getMinimum());
```

Comment:
This block is almost unreadable... can you do two if (isDecimal)/else with both min/max inside?

Reply:
Indeed, better to perform the string parsing in the same if sentence, applying the change.